cloud/terraform/base_config_builder.py

```diff
@@ -38,9 +38,21 @@ def create_resource_name(self, resource_names_combination, separator='-') -> str
         :param separator: String to be used as a separator of the items passed in resource_names_combination
         :return: String composed of a prefix, resource names combination and a suffix with random numbers.
         """
-        combinations = [self.resource_name_prefix]
-        combinations.extend(resource_names_combination)
-        combinations.append(self.get_random_numbers())
+        tf_resource_name_max_chars = 63
+        random_num = self.get_random_numbers()
+
+        combined_name = separator.join(resource_names_combination).replace(f'{self.resource_name_prefix}-', '')
+        combined_name_full_length = len(separator.join([self.resource_name_prefix, combined_name, random_num]))
+
+        if combined_name_full_length > tf_resource_name_max_chars:
+            chars_to_strip = combined_name_full_length - tf_resource_name_max_chars
+            combined_name = combined_name[0:-chars_to_strip]
+
+        combinations = [
+            self.resource_name_prefix,
+            combined_name,
+            random_num
+        ]
 
         return separator.join(combinations)
```

Comment:
Why not use a temporary variable and move the name truncation inside the `return` statement ? 

If you mean to keep the prefix but still struncate the rest of the name then why not construct the part after the prefix and truncate it before return ?

Reply:
This question is still valid. From what I can tell the entire function is almost equivalent to 

```
        combinations = [self.resource_name_prefix]
        combinations.extend(resource_names_combination)
        combinations.append(self.get_random_numbers())

        return separator.join(combinations)[0:63]
```

Given the existing doc-string
```
        :param resource_names_combination: List of keywords, names or IDs that you want to include in the resource name.
                                           Example: ['westeurope', 'vm']
```

I don't think that the middle bits are the ones which are too long so it doesn't really make any difference if characters are removed from the middle or from the end. 

I'm not quite sure what you are trying to do by changing this function that much, hoever it feels to me that in practice it only needs a very small change - the string slice operation at the end of the `return` statement.  Adding ~ 5 chars instead of unnecessarily complicating the logic of this function seems like a better option to me.

Reply:
Hello @atodorov , let me explain:
Sometimes we pass long names here, and if we don't truncate the "middle" part of it, it may end up with a duplicated resource name for terraform and making the provisioning fail. I encountered this issue by myself by running things locally and the name can sometimes be long enough. If we just truncate after the concatenation, the random numbers at the end that provide the unique identifier as a resource name,, may be truncated as well and that's when we encounter the "duplicated resource" issue.

In the end, we should a void super long names because they are not very user-friendly for debugging, however some names are taken from the image metadata itself, such as the vhd blob filename or what's defined in resources.json as image name, etc. We cannot control that.

Reply:
Thanks for explanation. This now makes more sense.